src/backend/cdb/executormgr.c

```diff
@@ -864,29 +864,42 @@ addOneOption(PQExpBufferData *buffer, struct config_generic * guc)
 			{
 				struct config_string *sguc = (struct config_string *) guc;
 				const char *str = *sguc->variable;
-				int			j,
-							start,
-							end;
-				char		temp[1024];
+				unsigned int	 j, start, size;
+				char			*temp;
 
-				end = strlen(str);
+				size = 256;
```

Comment:
256 does not have special meaning. It is just a rough initial size so that realloc() is seldom called.